osbuild/util/linux.py

```diff
@@ -0,0 +1,90 @@
+"""Linux API Access
+
+This module provides access to linux system-calls and other APIs, in particular
+those not provided by the python standard library. The idea is to provide
+universal wrappers with broad access to linux APIs. Convenience helpers and
+higher-level abstractions are beyond the scope of this module.
+
+In some cases it is overly complex to provide universal access to a specifc
+API. Hence, the API might be restricted to a reduced subset of its
+functionality, just to make sure we can actually implement the wrappers in a
+reasonable manner.
+"""
+
+
+import array
+import fcntl
+
+
+__all__ = [
+    "ioctl_get_immutable",
+    "ioctl_toggle_immutable",
+]
+
+
+# NOTE: These are wrong on at least ALPHA and SPARC. They use different
+#       ioctl number setups. We should fix this, but this is really awkward
+#       in standard python.
+#       Our tests will catch this, so we will not accidentally run into this
+#       on those architectures.
+FS_IOC_GETFLAGS = 0x80086601
+FS_IOC_SETFLAGS = 0x40086602
+
+FS_IMMUTABLE_FL = 0x00000010
+
+
+def ioctl_get_immutable(fd: int):
+    """Query FS_IMMUTABLE_FL
+
+    This queries the `FS_IMMUTABLE_FL` flag on a specified file.
+
+    Arguments
+    ---------
+    fd
+        File-descriptor to operate on.
+
+    Returns
+    -------
+    bool
+        Whether the `FS_IMMUTABLE_FL` flag is set or not.
+
+    Raises
+    ------
+    OSError
+        If the underlying ioctl fails, a matching `OSError` will be raised.
+    """
+
+    assert isinstance(fd, int) and fd >= 0
```

Comment:
I would still argue that `assert` is for asserting internal state and not checking of arguments, which I think should raise `ValueError`, but if you strongly prefer that I wont fuss around too much.

Reply:
From the google python [style guide](http://google.github.io/styleguide/pyguide.html):

> For example, raise a ValueError to indicate a programming mistake like a violated precondition (such as if you were passed a negative number but required a positive one). Do not use assert statements for validating argument values of a public API. assert is used to ensure internal correctness, not to enforce correct usage nor to indicate that some unexpected event occurred. If an exception is desired in the latter cases, use a raise statement. 

Not that we have to follow it, I just happen to agree in that case :smirk_cat:

Reply:
Fixed.